support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/ticket/code/OAuth20DefaultOAuthCodeFactory.java

```diff
@@ -79,7 +82,7 @@ public OAuth20Code create(final Service service,
             expirationPolicyToUse, ticketGrantingTicket, scopes,
             codeChallenge, codeChallengeMethod, clientId,
             requestClaims, responseType, grantType);
-        if (ticketGrantingTicket != null) {
+        if (casPoperties.getLogout().isRemoveDescendantTickets() && ticketGrantingTicket != null) {
```

Comment:
Suggestion: let's use a new setting:

```properties
cas.ticket.tgt.core.trackDescendantTickets=true
```

Similar to how `onlyTrackMostRecentSession` is defined.

Reply:
I close this one. I will use a local customisation for the 6.6.x branch.